### ISSUE TITLE
Core Plot does not compile on Xcode 9

### ISSUE THREAD

**Issue by LeoNatan** [core-plot/core-plot#339]
This is due to nullability mismatch, stricter checking in Xcode and treat warnings as errors. I think this is a correct error.

```objc
CPTNumberArray *yValues = @[[self cachedNumberForField:CPTTradingRangePlotFieldOpen recordIndex:idx],
                                [self cachedNumberForField:CPTTradingRangePlotFieldClose recordIndex:idx],
                                [self cachedNumberForField:CPTTradingRangePlotFieldHigh recordIndex:idx],
                                [self cachedNumberForField:CPTTradingRangePlotFieldLow recordIndex:idx]];
```

The error message:
> Implicit conversion from nullable pointer ‘NSNumber * _Nullable' to non-nullable pointer type 'ObjectType _Nonnull' (aka 'id')

`CPTNumberArray` is defined as:
```objc
typedef NSArray<NSNumber *> CPTNumberArray;
```

**Comment by eskroch** [comment]
Fixed on the `release-2.3` branch.

**Comment by tswast** [comment]
Thanks @eskroch! I don't see mention of the other branches in the `CONTRIBUTING.md`. It would be good to document how this project uses `master` versus how it uses the other branches.
